Fix/memory leak nogs gate button

Jhonattan2121 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented background timers from persisting after navigation, reducing battery use and avoiding unexpected follow-up actions.
  * Improved reliability of countdown updates and recheck behavior, minimizing missed or duplicated checks.

* **Refactor**
  * Stabilized internal callbacks to reduce unnecessary re-renders and enhance responsiveness.
  * Strengthened effect cleanup so timers are consistently cleared, increasing overall stability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sktbrd committed:
[WIP] - Add fallback URL handling for IFrame fidget (#1305)

Co-authored-by: Jesse Paterson <jesse.r.pat@gmail.com>

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
Merge branch 'Nounspace:canary' into canary

Jhonattan2121 committed:
fix: resolve memory leak in NogsGateButton component (#1330)

- Add cleanup function to useEffect for timer cleanup on unmount/re-run
- Fix missing checkForNogs dependency with useCallback memoization
- Add cleanup effect to clear all timers when component unmounts
- Prevents progressive memory leaks and performance degradation
- Eliminates 'state update on unmounted component' warnings

Fixes #1330

j-paterson commented on src/common/components/organisms/NogsGateButton.tsx:
```diff
@@ -89,22 +89,45 @@ const NogsGateButton = (props: ButtonProps) => {
+    setNogsRecheckTimerLength,
+    nogsRecheckTimerLength,
+    setNogsTimeoutTimer,
+  ]);
```
@Jhonattan2121 - check out some of the copilot comments

https://github.com/Nounspace/nounspace.ts/pull/1417/files/e0267bcad7cb0f47a3a9c8eadc264029314fffcd#r2342429535

Jhonattan2121 committed:
fix: optimize useCallback and useEffect dependencies

- Remove timer refs from useCallback to prevent infinite re-renders
- Remove useState setters from useEffect (stable references)
- Use empty dependency array [] for cleanup effect
